dotnet/Razorvine.Pyrolite/Pyrolite/Pickle/Unpickler.cs

```diff
@@ -93,24 +70,45 @@ public class Unpickler : IDisposable {
 	 * @return the reconstituted object hierarchy specified in the file.
 	 */
 	public object load(Stream stream) {
-		input = stream;
-		stack = new UnpickleStack();
-		while (true) {
-			byte key = PickleUtils.readbyte(input);
-			object value = dispatch(key);
-			if(value != NO_RETURN_VALUE)
-				return value;
-		}
-	}
+        stack = new UnpickleStack();
+        var unpickler = new Unpickler<StreamReader>(new StreamReader(stream), memo, stack, this);
```

Comment:
Similarly here:
```C#
var unpickler = stream is MemoryStream ms && ms.TryGetBuffer(out ArraySegment array) ?
    new Unpickler<ArrayReader>(new ArrayReader(array.Array, array.Offset), memo, stack, this) :
    new Unpickler<StreamReader>(new StreamReader(stream), memo, stack, this);
```
?

new Unpickler<StreamReader>(new StreamReader(stream), memo, stack, this);

Reply:
good point, however, I think that all unit tests for `Stream` use `MemoryStream` and if I would apply this change I would loose the test coverage for `Stream` (all tests would test the Array code path)